MVVM/app/src/main/java/com/lee989898/mvvm/MainViewModel.kt

```diff
@@ -0,0 +1,55 @@
+package com.lee989898.mvvm
+
+import android.service.autofill.Validators.and
+import android.util.Patterns
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MediatorLiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import java.util.regex.Pattern
+
+class MainViewModel : ViewModel() {
+    private val email = MutableLiveData("")
+    private val pwd = MutableLiveData("")
+    private val emailCheck = MutableLiveData(false)
+    private val pwdCheck = MutableLiveData(false)
+    private val loginButton: MediatorLiveData<Boolean> = MediatorLiveData()
+    private val loginSuccess = MutableLiveData(false)
+
+    init {
+        loginButton.addSource(emailCheck){
+            loginButton.value = _loginButton()
+        }
+        loginButton.addSource(pwdCheck){
+            loginButton.value = _loginButton()
+        }
+    }
+
+    fun setEmail(setEmail: String){
+        email.value = setEmail
+    }
+
+    fun loginSuccess(){
+        loginSuccess.value = true
+    }
+
+    fun _loginButton(): Boolean{
```

Comment:
```suggestion
    fun getEnabledLoginButton(): Boolean {
```

omg.. Reformat Code는 무족권입니다!!!!
이미 알고계실 것 같지만 윈도우 기준 `ctrl +alt + L`

함수명의 경우 보통 동사로 시작하는 것이 컨 벤 션!
사실 이 함수명도 완전 마음에 들지는 않지만,, **getEnabledLoginButton** 제안해 봅니다 👩‍🎤

Reply:
함수명도 바꾸겠습니다!!